var/spack/repos/builtin/packages/gperftools/package.py

```diff
@@ -22,4 +22,19 @@ class Gperftools(AutotoolsPackage):
     version('2.4', sha256='982a37226eb42f40714e26b8076815d5ea677a422fb52ff8bfca3704d9c30a2d')
     version('2.3', sha256='093452ad45d639093c144b4ec732a3417e8ee1f3744f2b0f8d45c996223385ce')
 
-    depends_on("unwind")
+    variant('sized_delete', default=False, description="Build sized delete operator")
+    variant('dynamic_sized_delete_support', default=False, description="Try to build run-time switch for sized delete operator")
+    variant('debugalloc', default=True, description="Build versions of libs with debugalloc")
+    variant('libunwind', default=True, description="Enable libunwind linking")
+
+    depends_on("unwind", when="+libunwind")
+
+    def configure_args(self):
+        args = []
+        for opt in ('sized_delete', 'dynamic_sized_delete_support'):
+            args.append('--{0}-{1}'.format('enable' if self.spec.variants[opt].value else 'disable', opt.replace('_', '-')))
```

Comment:
Personally I prefer a more declarative way of writing this instead of loops -- are we missing a enable or disable helper maybe?

Reply:
@haampie maybe a flag for `enable_or_disable` to translate `_` to `-` when formatting option that is further passed to `./configure`?

Reply:
Or, something along these lines https://github.com/spack/spack/pull/25841#issuecomment-922264365

Reply:
Yeah, or something simpler like `enable_or_disable('autotools_option', variant='spack_variant')`

Reply:
But it's not a blocker for this PR!

Reply:
Maybe. IMHO, it's just a question of where to put that value (inside `variant` or inside `_activate_or_not`)

Reply:
I'd say inside of `_activate_or_not`. The arguments passed to the configure script (which may change from autotools to cmake over time, by the way) shouldn't be part of the variant definition itself.